game-core/src/main/java/games/strategy/triplea/delegate/UnitComparator.java

```diff
@@ -107,6 +107,11 @@ private UnitComparator() {}
         return (route != null && route.isLoad()) ? left2.compareTo(left1) : left1.compareTo(left2);
       }
 
+      // Sort units by type first.
+      if (!u1.getType().equals(u2.getType())) {
+        return Integer.compare(u1.getType().hashCode(), u2.getType().hashCode());
+      }
+
```

Comment:
Would you mind explaining why we end up comparing hash codes?

I mean I see that this is being used a couple of lines below as well, but I don't understand it there either.

Reply:
The idea is that if you have 100 infantry and 100 tanks, then if the 100 infantry are all equivalent (same unit type, same movement left), we can skip all 100 instead of skipping one at a time since if one can't move, the others can't either.

Reply:
Ah ok, makes sense. I'd still like @ron-murhammer to have a second look at this 👍